api/src/services/listing.service.ts

```diff
@@ -1152,6 +1154,33 @@ export class ListingService implements OnModuleInit {
   */
   async update(dto: ListingUpdate, requestingUser: User): Promise<Listing> {
     const storedListing = await this.findOrThrow(dto.id, ListingViews.details);
+    const isNonAdmin = !requestingUser?.userRoles?.isAdmin;
+    const isActiveListing = dto.status === ListingsStatusEnum.active;
+
+    //check if the user has permission to edit dates
+    if (isNonAdmin && isActiveListing) {
+      const lotteryEvent = dto.listingEvents?.find(
+        (event) => event?.type === ListingEventsTypeEnum.publicLottery,
+      );
+      const storedLotteryEvent = storedListing.listingEvents?.find(
+        (event) => event?.type === ListingEventsTypeEnum.publicLottery,
+      );
+
+      if (
+        checkIfDatesChanged(
+          lotteryEvent,
+          storedLotteryEvent,
+          dto,
+          storedListing.applicationDueDate?.toISOString(),
+          storedListing.reviewOrderType,
+        )
+      ) {
+        throw new HttpException(
+          'You do not have permission to edit dates',
+          403,
+        );
+      }
+    }
```

Comment:
Nit: can we move this after the permissions check at line 1185? We should verify they have access to this listing before this imo